content/docs/intro/concepts/_index.md

```diff
@@ -9,49 +9,292 @@ menu:
 aliases: ["/docs/reference/concepts/"]
 ---
 
-Pulumi is a modern infrastructure as code platform. It includes a CLI, runtime, libraries, and a hosted service that, working together, deliver a robust way of provisioning, updating, and managing cloud infrastructure. Instead of YAML or a domain-specific language (DSL), Pulumi leverages existing, familiar programming languages, including TypeScript, JavaScript, Python, Go, and .NET, and their native tools, libraries, and package managers.
+Pulumi is a modern infrastructure as code platform. It includes a CLI, runtime, libraries, and a hosted service working together to deliver a robust way of provisioning, updating, and managing cloud infrastructure.
 
-This section will teach you the core concepts behind Pulumi, including how to author your Pulumi programs, the CLI you will use to deploy those programs, how projects (and their related concept, "stacks") help you to organize your environments, and how Pulumi works under the hood.
+Instead of YAML or a domain-specific language (DSL), Pulumi leverages existing programming languages---TypeScript, JavaScript, Python, Go, and .NET---and their native tools, libraries, and package managers to interact with cloud resources through the Pulumi SDK.
 
 > If this is your first time using Pulumi, you likely want to begin with [the Getting Started guide]({{< relref "/docs/get-started" >}}) for your cloud of choice. It will walk you through an [AWS]({{< relref "/docs/get-started/aws" >}}), [Azure]({{< relref "/docs/get-started/azure" >}}), [GCP]({{< relref "/docs/get-started/gcp" >}}), or [Kubernetes]({{< relref "/docs/get-started/kubernetes" >}}) deployment from start to finish.
 
-The following topics describe the core concepts behind Pulumi in more detail:
+## Programming Model Overview {#programmingmodel}
+
+This diagram illustrates the structure and major components of the Pulumi programming model.
+
+![Pulumi programming model diagram.](/images/docs/pulumi-programming-model-diagram.svg)
+
+*Programs* use existing programming languages to define how your infrastructure should be deployed. After writing a program, you run the Pulumi CLI command `up`, which executes the program and determines the desired infrastructure state for all resources declared.
+
+*Resources* represent a type of infrastructure that you declare in your program. Resources have properties that correspond to the desired state of your infrastructure and can be used as *inputs and outputs* that represent dependencies between resources.
+
+_Stacks_ are isolated, configurable instances of a Pulumi program. Because of their isolated nature, you can think of stacks are similar to deployment environments where your Pulumi program runs.
+
+The program, stacks, and other metadata reside in a *Project* directory.
+
+To illustrate these concepts, the following program shows how to create an AWS EC2 security group and an EC2 instance that uses it:
```

Comment:
If we want to use this to illustrate the concepts above, we may also want to say something about how we can deploy multiple instances of this program to different stacks.